File: java_projects/glue_code/joda-money/src/main/java/org/joda/money/format/SignedPrinterParser.java
package org.joda.money.format;

import java.io.IOException;
import java.io.Serializable;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;
import org.joda.money.BigMoney;
import org.joda.money.ContextInitializer;
import org.joda.money.ExceptionHandler;

final class SignedPrinterParser implements MoneyPrinter, MoneyParser, Serializable {
  private static final long serialVersionUID = 1L;
  private static Value clz =
      ContextInitializer.getPythonClass("/format/SignedPrinterParser.py", "SignedPrinterParser");
  private Value obj;

  public SignedPrinterParser(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public String toString() {
    //
    // return "PositiveZeroNegative(" + whenPositive + "," + whenZero + "," + whenNegative + ")";
    //

    // TODO: Check the type mapping below!
    return this.obj.invokeMember("toString").as(String.class);
  }

  public void parse(MoneyParseContext context) {
    //
    // MoneyParseContext positiveContext = context.createChild();
    // whenPositive.getPrinterParser().parse(positiveContext);
    // MoneyParseContext zeroContext = context.createChild();
    // whenZero.getPrinterParser().parse(zeroContext);
    // MoneyParseContext negativeContext = context.createChild();
    // whenNegative.getPrinterParser().parse(negativeContext);
    // MoneyParseContext best = null;
    // if (!positiveContext.isError()) {
    // best = positiveContext;
    // }
    // if (!zeroContext.isError()) {
    // if (best == null || zeroContext.getIndex() > best.getIndex()) {
    // best = zeroContext;
    // }
    // }
    // if (!negativeContext.isError()) {
    // if (best == null || negativeContext.getIndex() > best.getIndex()) {
    // best = negativeContext;
    // }
    // }
    // if (best == null) {
    // context.setError();
    // } else {
    // context.mergeChild(best);
    // if (best == zeroContext) {
    // if (context.getAmount() == null
    // || context.getAmount().compareTo(BigDecimal.ZERO) != 0) {
    // context.setAmount(BigDecimal.ZERO);
    // }
    // } else if (best == negativeContext
    // && context.getAmount().compareTo(BigDecimal.ZERO) > 0) {
    // context.setAmount(context.getAmount().negate());
    // }
    // }
    //

    this.obj.invokeMember("parse", context);
  }

  public void print(MoneyPrintContext context, Appendable appendable, BigMoney money)
      throws IOException {
    try {
      //
      // MoneyFormatter fmt =
      // (money.isZero() ? whenZero : money.isPositive() ? whenPositive : whenNegative);
      // fmt.getPrinterParser().print(context, appendable, money);
      //

      this.obj.invokeMember("print", context, appendable, money);
    } catch (PolyglotException e) {
      // TODO: Handle IOException
      throw (IOException) ExceptionHandler.handle(e, "SignedPrinterParser.print");
    }
  }

  SignedPrinterParser(
      MoneyFormatter whenPositive, MoneyFormatter whenZero, MoneyFormatter whenNegative) {
    //
    // this.whenPositive = whenPositive;
    // this.whenZero = whenZero;
    // this.whenNegative = whenNegative;
    //

    this.obj = clz.invokeMember("__init__", whenPositive, whenZero, whenNegative);
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/joda-money/src/main/org/joda/money/format/SignedPrinterParser.py
from __future__ import annotations
from io import IOBase
import io
import os
from src.main.org.joda.money.BigMoney import *
from src.main.org.joda.money.format.MoneyFormatter import *
from src.main.org.joda.money.format.MoneyParseContext import *
from src.main.org.joda.money.format.MoneyParser import *
from src.main.org.joda.money.format.MoneyPrintContext import *
from src.main.org.joda.money.format.MoneyPrinter import *
from src.main.org.joda.money.format.MultiPrinterParser import *


class SignedPrinterParser(MoneyParser, MoneyPrinter):

    __whenNegative: MoneyFormatter = None
    __whenZero: MoneyFormatter = None
    __whenPositive: MoneyFormatter = None

    __serialVersionUID: int = 1

    def toString(self) -> str:

        return (
            "PositiveZeroNegative("
            + self.__whenPositive
            + ","
            + self.__whenZero
            + ","
            + self.__whenNegative
            + ")"
        )

    def parse(self, context: MoneyParseContext) -> None:

        positive_context = context.createChild()
        self.__whenPositive.getPrinterParser().parse(positive_context)

        zero_context = context.createChild()
        self.__whenZero.getPrinterParser().parse(zero_context)

        negative_context = context.createChild()
        self.__whenNegative.getPrinterParser().parse(negative_context)

        best = None

        if not positive_context.isError():
            best = positive_context

        if not zero_context.isError():
            if best is None or zero_context.getIndex() > best.getIndex():
                best = zero_context

        if not negative_context.isError():
            if best is None or negative_context.getIndex() > best.getIndex():
                best = negative_context

        if best is None:
            context.setError()
        else:
            context.mergeChild(best)
            if best == zero_context:
                if (
                    context.getAmount() is None
                    or context.getAmount().compareTo(BigDecimal.ZERO) != 0
                ):
                    context.setAmount(BigDecimal.ZERO)
            elif (
                best == negative_context
                and context.getAmount().compareTo(BigDecimal.ZERO) > 0
            ):
                context.setAmount(context.getAmount().negate())

    def print(
        self, context: MoneyPrintContext, appendable: io.TextIOBase, money: BigMoney
    ) -> None:

        fmt = (
            self.__whenZero
            if money.isZero()
            else self.__whenPositive if money.isPositive() else self.__whenNegative
        )
        fmt.getPrinterParser().print(context, appendable, money)

    def __init__(
        self,
        whenPositive: MoneyFormatter,
        whenZero: MoneyFormatter,
        whenNegative: MoneyFormatter,
    ) -> None:

        self.__whenPositive = whenPositive
        self.__whenZero = whenZero
        self.__whenNegative = whenNegative
